js/src/sdk/models/Entity.ts

```diff
@@ -60,35 +79,12 @@ export class Entity {
           },
         });
       }
-      let connectedAccount = null;
-      if (connectedAccountId) {
-        connectedAccount = await this.connectedAccounts.get({
-          connectedAccountId: connectedAccountId,
-        });
-      } else {
-        const connectedAccounts = await this.connectedAccounts.list({
-          //@ts-ignore
-          user_uuid: this.id,
-          appNames: action.appKey,
-          status: "ACTIVE",
+      const connectedAccount = this.getConnection({app: action.appKey, connectedAccountId});
+  
+      if (!connectedAccount) {
+        CEG.throwCustomError(SDK_ERROR_CODES.SDK.NO_CONNECTED_ACCOUNT_FOUND, {
```

Comment:
The error message for connection not found could be more descriptive. Consider adding more context:

```typescript
CEG.throwCustomError(SDK_ERROR_CODES.SDK.NO_CONNECTED_ACCOUNT_FOUND, {
  description: `No active connection found for app='${action.appKey}' and entity='${this.id}'. Please ensure a valid connection exists and has the required permissions.`,
});